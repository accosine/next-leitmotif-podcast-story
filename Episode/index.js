import React from "react";
import MDX from "@mdx-js/runtime";
import Attachment from "../Attachment";

const Episode = ({
  audiosnippet,
  bitlength,
  content,
  cta,
  date,
  draft,
  duration,
  episode,
  guid,
  image,
  nameMark,
  number,
  podcasts,
  poster,
  subtitle,
  title,
  video,
}) => (
  <>
    <amp-story-page id={`page${number}`} background-audio={audiosnippet}>
      {!!video && (
        <amp-story-grid-layer template="fill">
          <amp-video
            autoplay=""
            loop=""
            width="720"
            height="1280"
            poster={`/${poster}`}
            layout="responsive"
          >
            <source src={`/${video}`} type="video/mp4" />
          </amp-video>
        </amp-story-grid-layer>
      )}
      {!!!video && !!poster && (
        <amp-story-grid-layer template="fill">
          <amp-img
            src={`/${poster}`}
            width="720"
            height="1280"
            layout="responsive"
          ></amp-img>
        </amp-story-grid-layer>
      )}
      {!!nameMark && !!title && (
        <amp-story-grid-layer template="vertical">
          <div
            className="episodetitle content"
            animate-in="fly-in-top"
            animate-in-duration=".5s"
          >
            <h1>{title}</h1>
          </div>
          <div
            className="imgcontainer content"
            animate-in="twirl-in"
            animate-in-duration="1s"
          >
            <amp-img
              src={`/${nameMark}`}
              width="600"
              height="600"
              layout="responsive"
            ></amp-img>
          </div>
        </amp-story-grid-layer>
      )}
      {!!content && (
        <amp-story-grid-layer template="vertical" className="bottom padbutt">
          <div
            className="episodedesc"
            animate-in="fly-in-bottom"
            animate-in-duration="0.9s"
          >
            <MDX>{content}</MDX>
          </div>
        </amp-story-grid-layer>
      )}
      <Attachment podcasts={podcasts} cta={cta} />
    </amp-story-page>
  </>
);

export default Episode;
