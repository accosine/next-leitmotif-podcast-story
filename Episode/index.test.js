import React from "react";
import "@testing-library/jest-dom/extend-expect";
import { render } from "@testing-library/react";
import Episode from "./index";
import {
  audiosnippet,
  content,
  nameMark,
  podcasts,
  poster,
  title,
  video,
  number,
} from "../tests/fixtures.js";

const episodeProps = {
  audiosnippet,
  content,
  nameMark,
  number,
  podcasts,
  poster,
  title,
  video,
};
describe("Episode", () => {
  it("renders the Episode component", () => {
    const { container } = render(<Episode {...episodeProps} />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
