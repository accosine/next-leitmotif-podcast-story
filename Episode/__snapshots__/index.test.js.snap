// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Episode renders the Episode component 1`] = `
<amp-story-page
  background-audio="fakeAudioSnippet.mp3"
  id="page00"
>
  <amp-story-grid-layer
    template="fill"
  >
    <amp-video
      autoplay=""
      height="1280"
      layout="responsive"
      loop=""
      poster="/fakePoster.jpg"
      width="720"
    >
      <source
        src="/fakeVideo.mp4"
        type="video/mp4"
      />
    </amp-video>
  </amp-story-grid-layer>
  <amp-story-grid-layer
    template="vertical"
  >
    <div
      animate-in="fly-in-top"
      animate-in-duration=".5s"
      class="episodetitle content"
    >
      <h1>
        fakeTitle
      </h1>
    </div>
    <div
      animate-in="twirl-in"
      animate-in-duration="1s"
      class="imgcontainer content"
    >
      <amp-img
        height="600"
        layout="responsive"
        src="/fakeNameMark"
        width="600"
      />
    </div>
  </amp-story-grid-layer>
  <amp-story-grid-layer
    classname="bottom padbutt"
    template="vertical"
  >
    <div
      animate-in="fly-in-bottom"
      animate-in-duration="0.9s"
      class="episodedesc"
    >
      <h1>
        fakeHeadline
      </h1>
      <p>
        with super fake content
      </p>
    </div>
  </amp-story-grid-layer>
  <amp-story-page-attachment
    data-cta-text="Podcast jetzt abonnieren"
    layout="nodisplay"
  >
    <div
      class="abo"
    >
      <h2>
        Abonnieren
      </h2>
      <p
        class="attachment"
      >
        NFFS gibts auf folgenden Podcast-Plattformen
      </p>
      <p
        class="attachment"
      >
        <a
          href="fakePodcastShow1"
        >
          <amp-img
            height="68"
            layout="responsive"
            src="fakePodcastImage1.jpg"
            width="268"
          />
        </a>
      </p>
      <p
        class="attachment"
      >
        <a
          href="fakePodcastShow2"
        >
          <amp-img
            height="68"
            layout="responsive"
            src="fakePodcastImage2.jpg"
            width="268"
          />
        </a>
      </p>
      <p
        class="attachment"
      >
        Einfach draufklicken und in der App abonnieren um nichts zu verpassen!
      </p>
    </div>
  </amp-story-page-attachment>
</amp-story-page>
`;
